File: pertemuan3/blockBit.cpp
#include <iostream>
#include <vector>
using namespace std;

int N, K;
vector <vector<int>> result;
vector <int> blockBit;

void placeBit(int iterator, int remainK){
    if (iterator == N) {
        if (remainK == 0){
            result.push_back(blockBit);
            return;
        }
    }
    else {
        if (remainK > (N - iterator + 1) / 2) {
            return;
        }

        blockBit[iterator] =  0;
        placeBit(iterator+1,remainK);

        bool isSafe = (iterator == 0 || blockBit[iterator-1] == 0);

        if (isSafe && remainK > 0){
            blockBit[iterator] = 1;
            placeBit(iterator+1, remainK-1);
            blockBit[iterator] = 0;
        }
    }
}


int main () {
    cin >> N >> K;
    blockBit.assign(N,0);

    placeBit(0,K);

    if (result.empty()) {
        cout<<-1<<endl;
        return 0;
    }

    for (const auto& i: result) {
        for (int j = 0; j < N;j++) {
            cout << i[j] << " ";
        }
        cout << endl;
    }

}

File: pertemuan3/blockBit.py

N, K = map(int, input().split())

result = []
blockPita = [0]*N

def placeBit(iterator, remainK) :
    if iterator == N:
        if remainK == 0 :
            result.append(blockPita[:])
        return 
    else:
        if remainK > (N-iterator+1) // 2: 
            return 
        
        blockPita[iterator] = 0
        placeBit(iterator+1,remainK)


        isSafe = (iterator == 0 or blockPita[iterator-1] == 0)
        if isSafe and remainK > 0:
            blockPita[iterator] = 1
            placeBit(iterator+1,remainK-1)
            blockPita[iterator] = 0

placeBit(0,K)

if result == []:
    print(-1)
else :
    for i in result:
        for j in i:
            print(j,end=" ")
        print()